### ISSUE TITLE
Using the 5500PCU (USB) interface

### ISSUE THREAD

**Issue by vbastos** [micolous/cbus#3]
Hi there,

My electrician is currently installing a cbus system in my house and I'm looking at integrating with something like home-assistant.io. I am just wondering whether this code is still relevant to the current cbus systems and whether you might have any tips/guidance for the integration. Is the USB module ok to use or is it better to use the RS232 because you mention that the kernel module "should" function...

Thanks

**Comment by micolous** [comment]
To my knowledge, the code should work with any CBus system following CBus2 protocol documentation put out by Clipsal.  Provided Clipsal haven't changed it, it should still work.

The 5500PCU should be fine to use, but operating system support for it is in the scope of the Linux kernel, not this project.  The 5500PCU is just a 5500PC with a USB-Serial adaptor soldered on to it.

I've never actually tested it, as I only had a 5500PC (which I used with an external USB-serial adapter).  I got the information about the 5500PCU by poking at the Windows INF files supplied for the device, which I supplied to the kernel driver maintainer.

As for getting it running, basically everything runs around cdbusd -- this takes over your 5500PC(U) and provides a simple abstraction layer for lighting messages over DBus.

When you want remote/web control, then run sage as well, this gives you a WebSockets API which just pipes messages to DBus.

I've not attempted to integrate it with other software, as this code largely predates the numerous frameworks that have come out for home automation in the last couple of years.  Given that this codebase is Python 2, I'd suggest that you use the DBus API (and cdbusd) for integrating with home-assistant.io.

Given that the software isn't actually broken and supports all the feature set of CBus I wanted, and the installation of it that I maintain is now > 1,000 km from me, I haven't changed the repository in a long while.  This code base depends on Twisted which limits it's transition to Python 3.  However much of the actual protocol implementation does not depend on Twisted, so it may be possible to adapt that into something that is more tightly coupled to another home automation framework.

The time automation I run on that network uses cronjobs to achieve it.
